compiler/test/stdlib/path.test.gr

```diff
@@ -0,0 +1,165 @@
+import Path from "path"
+import Result from "result"
+import Option from "option"
+import List from "list"
+
+// test that a variety of properties hold for valid relative/absolute paths
+let relTests = [
+  (
+    "./dir/file.txt",
+    "./dir",
+    "./dir/file.txt",
+    Some("file.txt"),
+    Some("file"),
+    Some(".txt"),
+  ),
+  ("file", ".", "./file", Some("file"), Some("file"), None),
+  ("dir/../../", "../..", "..", None, None, None),
+  (".git", ".", "./.git", Some(".git"), Some(".git"), None),
+  (".", "..", ".", None, None, None),
+  ("..", "../..", "..", None, None, None),
+  ("../../file", "../..", "../../file", Some("file"), Some("file"), None),
+  ("./", "..", ".", None, None, None),
+  (
+    "dir//file.txt///",
+    "dir",
+    "./dir/file.txt",
```

Comment:
We don't want to keep a single trailing separator to indicate it is a directory path? I often waffle on these

Reply:
Sorry, I'm not exactly sure what you mean

Reply:
Sorry, I was wondering if we should keep a trailing slash on paths that are parsed with a trailing slash. Usually this is done by tools that need to track a path is exactly for a directory (since paths don't really encode that)

Reply:
Hmm, so are you saying to maintain some sort of information about whether a path is a directory or not? Then if it has a trailing slash when parsed it can be determined with certainty that it is a directory, whereas if it doesn't it is ambiguous?

Reply:
Yeah, that's what I'm wondering (if we should maintain that). It's something that I've needed many times in gulp

Reply:
Gotcha, do you think this idea of keeping the "directory-ness" of a path should be extended further into this library? For instance maybe encoding it in the type in the same way that absolute vs relative is to make the type something like `Path<Absolute, Directory>`. I can see some possible additional upsides of doing this, like only allowing `stem` and `extension` for `Path<a, File>`, `append` for `Path<a, Directory>`, etc.

Reply:
Oh! That's very interesting. How much more complicated do you think that'd be?

Reply:
I don't think that it would add a lot of complexity, the only changes I can foresee are the need for more parsing functions to capture each possible case e.g. `relativeFile`, `absoluteDirectory`, etc. Maybe also some functions that allow you to convert a file path to a directory path (less sure about the need for this though)?

Reply:
I added changes to accommodate this; let me know what you think